### PR TITLE
Button states

### PR DESCRIPTION
**Pull request by adebroux**
## Problem
We want buttons to have multiple configurable states.

## Solution
Add child components for 2 additional button states. Each state should have all the same props as the default state, in addition to props for the state name and the condition for the state to be shown. 

## Related PRs
https://github.com/AdaloHQ/frontend/pull/2835
https://github.com/AdaloHQ/runner/pull/876

### CODE REVIEW

**Comment by adebroux** [comment]
[Additional states for buttons](https://trello.com/c/jeTrrwt2/1564-additional-states-for-buttons)

**Review by michael-adalo — COMMENTED** [review]
this looks good, wonderfully simple.  I left just one comment, nothing major!

**Review by michael-adalo — APPROVED** [review]
✅

File: src/TextButton/TextButton.js [hunk 1]
```diff
@@ -113,17 +139,42 @@ export default class WrappedTextButton extends Component {
   }
 
   getAdditionalProps() {
-    let { type, shadow = true } = this.props
+    let { type = this.props.type, shadow = true } = this.getButtonState()
 
-    if (type === 'contained' && shadow) {
+    if (['contained', 'custom'].includes(type) && shadow) {
```
Comment (michael-adalo) [inline src/TextButton/TextButton.js:144]: nitpick; you could use something like `const RAISED_BUTTON_TYPES = new Set(['contained', 'custom'])` at the top of the file, then... 

```suggestion
    if (RAISED_BUTTON_TYPES.has(type) && shadow) {
```

File: src/TextButton/TextButton.js [hunk 2]
```diff
@@ -30,7 +30,16 @@ export default class WrappedTextButton extends Component {
   }
 
   getContainerStyles() {
-    let { type, primaryColor, borderRadius, sizing, icon, text } = this.props
+    const defaults = this.props
+    let {
```
Comment (michael-adalo) [inline src/TextButton/TextButton.js:34]: nitpick/question, could this be `const`?

File: src/TextButton/TextButton.js [hunk 3]
```diff
@@ -69,12 +88,19 @@ export default class WrappedTextButton extends Component {
   }
 
   getTextStyles() {
-    let { primaryColor, contrastColor, type, icon, styles, sizing, _fonts } =
-      this.props
+    const { _fonts, ...defaults } = this.props
+    let {
```
Comment (michael-adalo) [inline src/TextButton/TextButton.js:92]: nitpick/question, could this be `const`?

File: src/TextButton/TextButton.js [hunk 4]
```diff
@@ -69,12 +88,19 @@ export default class WrappedTextButton extends Component {
...
+      type = defaults.type,
+      icon = defaults.icon,
+      styles = defaults.styles,
+      sizing = defaults.sizing,
+    } = this.getButtonState()
 
     const textStyles = { fontWeight: '600' }
 
-    if (contrastColor && type === 'contained') {
+    if (contrastColor && ['contained', 'custom'].includes(type)) {
```
Comment (michael-adalo) [inline src/TextButton/TextButton.js:103]: same comment re. using the Set, but again, it's a nitpick

**Review by michael-adalo — APPROVED** [review]
✅ approved!  just a couple of nitpick-y comments

**Pull request merged**